pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1395,6 +1407,48 @@ func (r *ReconcileClusterDeployment) removeClusterDeploymentFinalizer(cd *hivev1
 	return nil
 }
 
+func (r *ReconcileClusterDeployment) releaseCustomization(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	if cd.Spec.ClusterPoolRef == nil || cd.Spec.ClusterPoolRef.CustomizationRef == nil {
+		return nil
+	}
+	customizationRef := cd.Spec.ClusterPoolRef.CustomizationRef
+	cdc := &hivev1.ClusterDeploymentCustomization{}
+	err := r.Client.Get(context.TODO(), client.ObjectKey{Namespace: cd.Spec.ClusterPoolRef.Namespace, Name: customizationRef.Name}, cdc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			cdLog.WithField("Customization", customizationRef.Name).Info("customization not found, nothing to release")
+			controllerutils.DeleteFinalizer(cd, hivev1.FinalizerCustomizationRelease)
+			if err := r.Update(context.TODO(), cd); err != nil {
+				cdLog.WithError(err).WithField("Customization", customizationRef.Name).Error("failed to update ClusterDeployment")
```

Comment:
Missing error return
```suggestion
				cdLog.WithError(err).WithField("Customization", customizationRef.Name).Error("failed to update ClusterDeployment")
				return err
```